aerogear/app-metrics-operator
# AEROGEAR-9184

**JIRA**
https://issues.jboss.org/browse/AEROGEAR-9184

**Tasks**
- [x] Add some dependencies in order to ad the new api for the a MobileSecurityApp
- [x] Generate the stuff
- [x] Have the operator respond to the creation of the app cr, aka create the required SDK config
- [x] Put the right stuff in the Confg Map
- [x] Edit - Have the config map be created if the cr is in a namespace which is provided by the APP_NAMESPACE variable.

**Verification Steps**

`make cluster/clean`
`make cluster/prepare`
`make install`

Run the following to verify the _correct_ namespace:
`make example-app/apply`
Verify that the sdk config map is created in the `app-metrics-apps` namespace with the name `<cr-app-name>-metrics`
Verify the contents of the config map is `url` : `the route for the metrics service`

Run the following to verify the _incorrect_ namespace:
`make example-app/delete`
Change the APP_NAMESPACES variable in the Makefile to `app-metrics`
`make example-app/apply`

Verify the cr is created in the `app-metrics` ns
Verify that the sdk config map is **not** created in the `app-metrics` namespace
Verify that the operator produces a log with the following content `"app-metrics/example-app","error":"The app cr example-app was created in a namespace which is not present in the APP_NAMESPACES environment variable provided to the operator","`
